src/pages/home/categories-list/categories-list.js

```diff
@@ -9,23 +10,27 @@ import {
   URL_QUERIES_NAME,
   countPerPage
 } from '../../../configs';
-
+import { getAllCategoriesQuery } from './operations/categories.queries';
+import errorOrLoadingHandler from '../../../utils/errorOrLoadingHandler';
 import { useStyles } from './categories-list.style';
+
 import './categories-carousel.css';
 
 const CategoriesList = () => {
-  const { categories, language, quantityPerPage } = useSelector(
-    ({ Categories, Language, Products }) => ({
-      categories: Categories.list,
-      language: Language.language,
-      quantityPerPage: Products.countPerPage
-    })
-  );
   const styles = useStyles();
+  const [categories, setCategories] = useState([]);
+  const { language, quantityPerPage } = useSelector(({ Language, Products }) => ({
+    language: Language.language,
+    quantityPerPage: Products.countPerPage
+  }));
 
-  const categoriesList = categories
-    ? categories
-      .map(({ _id, name, images }) => (
+  const { loading, error } = useQuery(getAllCategoriesQuery, {
+    onCompleted: (data) => setCategories(data.getAllCategories.items)
```

Comment:
If we are using the same endpoint in 2 components, and those are getting data at the same time, maybe we should use useContext, or take data in its parent and just pass data with props?

Reply:
To tell truth, it's not the same endpoint, there are 2 NEARLY similar queries named the same. and there are at least 2 more places in application, that should use similar endpoint.
However, we've been discussing this with my teammates yesterday, and agreed that there is no point to keep same code in two or more places of app. So that will be changed

Reply:
Ok, waiting